git/github/github.go

```diff
@@ -302,6 +318,89 @@ func (c *Client) GetCheckRunID(ctx context.Context, owner string, repo string, S
 	return nil, nil
 }
 
+// GetAllCheckRunsForRef returns all existing GitHub CheckRuns if a match for the Owner, Repo, SHA, and appID.
+func (c *Client) GetAllCheckRunsForRef(ctx context.Context, owner string, repo string, SHA string, appID int64) ([]*ghapi.CheckRun, error) {
+	filter := "all"
+
+	res, _, err := c.GetChecksService().ListCheckRunsForRef(
+		ctx,
+		owner,
+		repo,
+		SHA,
+		&ghapi.ListCheckRunsOptions{
+			AppID:  &appID,
+			Filter: &filter,
+		},
+	)
+
+	if err != nil {
+		return nil, err
```

Comment:
Can we be there little more explicit? Something like: "Unable to get checkRuns for repo/owner/SHA".

Reply:
Fixed. Thanks.